src/lune/builtins/luau.rs

```diff
@@ -0,0 +1,73 @@
+use mlua::prelude::*;
+use mlua::Compiler as LuaCompiler;
+
+use crate::lune::lua::table::TableBuilder;
+// use  as LuaNumber;
+
+pub fn create(lua: &'static Lua) -> LuaResult<LuaTable> {
+    TableBuilder::new(lua)?
+        .with_function("compile", compile_source)?
+        .with_function("load", load_source)?
+        .build_readonly()
+}
+
+fn compile_source<'a>(
+    lua: &'static Lua,
+    (source, options): (LuaString<'a>, Option<LuaTable<'a>>),
+) -> LuaResult<LuaString<'a>> {
+    let mut optimization_level = 1;
+    let mut coverage_level = 0;
+    let mut debug_level = 1;
+
+    if let Some(options) = options {
+        optimization_level = match options.raw_get("optimizationLevel")? {
+            LuaValue::Integer(val) => val as u8,
+            _ => optimization_level,
+        };
+
+        coverage_level = match options.raw_get("coverageLevel")? {
+            LuaValue::Integer(val) => val as u8,
+            _ => coverage_level,
+        };
+
+        debug_level = match options.raw_get("debugLevel")? {
+            LuaValue::Integer(val) => val as u8,
+            _ => debug_level,
+        };
+    };
+
+    let source_bytecode_bytes = LuaCompiler::default()
+        .set_optimization_level(optimization_level)
+        .set_coverage_level(coverage_level)
+        .set_debug_level(debug_level)
+        .compile(source);
+
+    match lua.create_string(source_bytecode_bytes) {
```

Comment:
I don't think matching is necessary here, `create_string` already returns a `LuaResult` for us

Reply:
Addressed in; https://github.com/filiptibell/lune/pull/82/commits/cd0a20b3712d6235181a0fd1375d9d1175866de2